seata1.4.0 启动报错

重启seata服务时，报错如下，导致无法启动
Caused by: java.lang.NullPointerException
	at java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:469)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
	at java.util.TimSort.sort(TimSort.java:220)
	at java.util.Arrays.sort(Arrays.java:1512)
	at java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:353)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:483)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
	at io.seata.server.storage.redis.store.RedisTransactionStoreManager.readBranchSessionByXid(RedisTransactionStoreManager.java:463)
	at io.seata.server.storage.redis.store.RedisTransactionStoreManager.readSession(RedisTransactionStoreManager.java:318)
	at io.seata.server.storage.redis.store.RedisTransactionStoreManager.lambda$readSession$2(RedisTransactionStoreManager.java:357)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)

--- a/changes/1.5.0.md
+++ b/changes/1.5.0.md
@@ -42,7 +42,7 @@ Seata 是一款开源的分布式事务解决方案，提供高性能和简单�
   - [[#3451](https://github.com/seata/seata/pull/3451)] 修复当不使用本地事务且设置自动提交为true时，全局锁竞争失败会使得rm退出全局事务，导致全局锁在rm重试时失效，数据被脏写
   - [[#3481](https://github.com/seata/seata/pull/3481)] 修复当 consul client 获取集群信息报错时会导致刷新任务中断
   - [[#3491](https://github.com/seata/seata/pull/3491)] 修复README.md文件中的拼写错误
-  - [[#3531](https://github.com/seata/seata/pull/3531)] 修复RedisTransactionStoreManager读取brachTransaction中的NEP
+  - [[#3531](https://github.com/seata/seata/pull/3531)] 修复RedisTransactionStoreManager读取brachTransaction中的NPE
 
   ### optimize： 
 

--- a/changes/1.5.0.md
+++ b/changes/1.5.0.md
@@ -42,7 +42,7 @@ Seata 是一款开源的分布式事务解决方案，提供高性能和简单�
   - [[#3451](https://github.com/seata/seata/pull/3451)] 修复当不使用本地事务且设置自动提交为true时，全局锁竞争失败会使得rm退出全局事务，导致全局锁在rm重试时失效，数据被脏写
   - [[#3481](https://github.com/seata/seata/pull/3481)] 修复当 consul client 获取集群信息报错时会导致刷新任务中断
   - [[#3491](https://github.com/seata/seata/pull/3491)] 修复README.md文件中的拼写错误
-
+  - [[#3531](https://github.com/seata/seata/pull/3531)] 修复RedisTransactionStoreManager读取brachTransaction中的NEP
 
   ### optimize： 
 
@@ -72,7 +72,6 @@ Seata 是一款开源的分布式事务解决方案，提供高性能和简单�
   - [[#3236](https://github.com/seata/seata/pull/3236)] 优化执行解锁操作的条件，减少不必要的store操作。
   - [[#3485](https://github.com/seata/seata/pull/3485)] 优化 ConfigurationFactory 中无用的try/catch
   - [[#3505](https://github.com/seata/seata/pull/3505)] 优化GlobalTransactionScanner类中无用的if判断
-  
 
   ### test
 
@@ -102,6 +101,7 @@ Seata 是一款开源的分布式事务解决方案，提供高性能和简单�
   - [13414850431](https://github.com/13414850431)
   - [github-ganyu](https://github.com/github-ganyu)
   - [xuande](https://github.com/xuande)
+  - [tanggen](https://github.com/tanggen)
 
 
 同时，我们收到了社区反馈的很多有价值的issue和建议，非常感谢大家。

--- a/changes/en-us/1.5.0.md
+++ b/changes/en-us/1.5.0.md
@@ -43,6 +43,7 @@
   - [[#3451](https://github.com/seata/seata/pull/3451)] fix set auto-commit to true when local transactions are not being used. Failure to compete for a lock causes the global transaction to exit, invaliding the global row lock and dirty writing of the data.
   - [[#3481](https://github.com/seata/seata/pull/3481)] fix seata node refresh failure because consul crash
   - [[#3491](https://github.com/seata/seata/pull/3491)] fix typo in README.md
+  - [[#3531](https://github.com/seata/seata/pull/3531)] fix the NEP of RedisTransactionStoreManager when get branch transactions
 
 
   ### optimize： 	
@@ -105,6 +106,7 @@
   - [13414850431](https://github.com/13414850431)
   - [github-ganyu](https://github.com/github-ganyu)
   - [xuande](https://github.com/xuande)
+  - [tanggen](https://github.com/tanggen)
   
 
   Also, we receive many valuable issues, questions and advices from our community. Thanks for you all.	

--- a/changes/en-us/1.5.0.md
+++ b/changes/en-us/1.5.0.md
@@ -43,7 +43,7 @@
   - [[#3451](https://github.com/seata/seata/pull/3451)] fix set auto-commit to true when local transactions are not being used. Failure to compete for a lock causes the global transaction to exit, invaliding the global row lock and dirty writing of the data.
   - [[#3481](https://github.com/seata/seata/pull/3481)] fix seata node refresh failure because consul crash
   - [[#3491](https://github.com/seata/seata/pull/3491)] fix typo in README.md
-  - [[#3531](https://github.com/seata/seata/pull/3531)] fix the NEP of RedisTransactionStoreManager when get branch transactions
+  - [[#3531](https://github.com/seata/seata/pull/3531)] fix the NPE of RedisTransactionStoreManager when get branch transactions
 
 
   ### optimize： 	

--- a/server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java
+++ b/server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java
@@ -453,7 +453,6 @@ public class RedisTransactionStoreManager extends AbstractTransactionStoreManage
             for (Object branchInfo : branchInfos) {
                 if (branchInfo != null) {
                     Map<String, String> branchInfoMap = (Map<String, String>) branchInfo;
-                    //maybe async_commit task has delete the map in redis,so make sure the map not null
                     Optional<BranchTransactionDO> branchTransactionDO =
                             Optional.ofNullable((BranchTransactionDO) BeanUtils.mapToObject(branchInfoMap, BranchTransactionDO.class));
                     branchTransactionDO.ifPresent(branchTransactionDOs::add);

--- a/server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java
+++ b/server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java
@@ -15,12 +15,7 @@
  */
 package io.seata.server.storage.redis.store;
 
-import java.util.ArrayList;
-import java.util.Comparator;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -452,9 +447,10 @@ public class RedisTransactionStoreManager extends AbstractTransactionStoreManage
             for (Object branchInfo : branchInfos) {
                 if (branchInfo != null) {
                     Map<String, String> branchInfoMap = (Map<String, String>) branchInfo;
-                    BranchTransactionDO branchTransactionDO =
-                            (BranchTransactionDO) BeanUtils.mapToObject(branchInfoMap, BranchTransactionDO.class);
-                    branchTransactionDOs.add(branchTransactionDO);
+                    //maybe async_commit task has delete the map in redis,so make sure the map not null
+                    Optional<BranchTransactionDO> branchTransactionDO =
+                            Optional.ofNullable((BranchTransactionDO) BeanUtils.mapToObject(branchInfoMap, BranchTransactionDO.class));
+                    branchTransactionDO.ifPresent(branchTransactionDOs::add);
                 }
             }
         }

--- a/server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java
+++ b/server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java
@@ -15,7 +15,13 @@
  */
 package io.seata.server.storage.redis.store;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;

